video_xblock/static/js/studio-edit.js

```diff
@@ -224,23 +225,30 @@ function StudioEditableXBlock(runtime, element) {
             }
         })
     };
-
-    var pushTranscript = function (lang, label, url){
+    /** Replaces an existing transcript to transcriptsValue or adds new */
+    var pushTranscript = function (lang, label, url, oldLang, $uploadButton){
```

Comment:
Please split this function into smaller one, move as much as possible to backend. 
Do not mess rendering and business logic in one function.